### PR TITLE
Do not crash on rule exceptions - #793 #944

### PR DESCRIPTION
**Pull request by arturbosch**

### CODE REVIEW

**Comment by schalkms** [comment]
@vanniktech good idea

**Comment by arturbosch** [comment]
Good idea! -> Done
Now printing the file path, detekt and java version plus the os name and stacktrace on rule exceptions!

File: detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/DebugUtils.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+fun whichOS(): String = System.getProperty("os.name")
+
+fun whichJava(): String = System.getProperty("java.runtime.version")
+
+fun whichDetekt(): String {
+	for (resource in Detektor::class.java.classLoader
+			.getResources("META-INF/MANIFEST.MF")) {
+		try {
+			Manifest(resource.openStream())
```
Comment (vanniktech) [inline detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/DebugUtils.kt]: Do we need to close the stream?

Comment (arturbosch) [reply detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/DebugUtils.kt]: Your right, thanks!

**Pull request merged**